lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/AccessControlRules.java

```diff
@@ -35,10 +35,10 @@ public AccessControlRules(
             @JsonProperty("session_properties") @JsonAlias("sessionProperties") Optional<List<SessionPropertyAccessControlRule>> sessionPropertyRules,
             @JsonProperty("functions") Optional<List<FunctionAccessControlRule>> functionRules)
     {
-        this.schemaRules = schemaRules.orElse(ImmutableList.of(SchemaAccessControlRule.ALLOW_ALL));
-        this.tableRules = tableRules.orElse(ImmutableList.of(TableAccessControlRule.ALLOW_ALL));
-        this.sessionPropertyRules = sessionPropertyRules.orElse(ImmutableList.of(SessionPropertyAccessControlRule.ALLOW_ALL));
-        this.functionRules = functionRules.orElse(ImmutableList.of(FunctionAccessControlRule.ALLOW_ALL));
+        this.schemaRules = schemaRules.orElseGet(() -> ImmutableList.of(SchemaAccessControlRule.ALLOW_ALL));
```

Comment:
(X) Object creation

Reply:
Does the null check matter?

(reverted anyway)